build.zig

```diff
@@ -0,0 +1,1484 @@
+        orca_tool_exe.linkFramework("SystemConfiguration");
+        orca_tool_exe.linkFramework("Security");
+    }
+
+    orca_tool_exe.step.dependOn(&curl_lib.step);
+    orca_tool_exe.step.dependOn(&z_lib.step);
+    orca_tool_exe.linkLibC();
+
+    const build_orca_tool: *Build.Step.InstallArtifact = b.addInstallArtifact(orca_tool_exe, .{});
+
+    var run_orca_tool: *Build.Step.Run = b.addRunArtifact(orca_tool_exe);
+    run_orca_tool.step.dependOn(&build_orca_tool.step);
+    if (b.args) |args| {
+        run_orca_tool.addArgs(args); // forwards args afer -- to orca cli, ex: zig build orca-tool -- update
+    }
+
+    const run_tool_step = b.step("orca-tool", "Run the Orca CLI tool");
+    run_tool_step.dependOn(&run_orca_tool.step);
+
+    /////////////////////////////////////////////////////////
+    // build_dependencies helper program
+
+    const build_deps_exe: *Build.Step.Compile = b.addExecutable(.{
+        .name = "build_dependencies",
+        .root_source_file = b.path("src/build/build_dependencies.zig"),
+        .target = target,
+        .optimize = .Debug,
+    });
+    // build_deps_exe.addIncludePath(b.path("src/ext/curl/include"));
+    // build_deps_exe.addIncludePath(b.path("src/ext/microtar"));
+    // build_deps_exe.linkLibrary(curl_lib);
+    // build_deps_exe.linkLibrary(z_lib);
+    // build_deps_exe.addCSourceFiles(.{
+    //     .files = &.{"src/ext/microtar/microtar.c"},
+    //     .flags = &.{},
+    // });
+
+    /////////////////////////////////////////////////////////
+    // angle build
+
+    var run_angle_build: *Build.Step.Run = b.addRunArtifact(build_deps_exe);
+    run_angle_build.addArg("--lib=angle");
+    run_angle_build.addArg(b.fmt("--sha={s}", .{shas.angle}));
+    run_angle_build.addArg(b.fmt("--intermediate={s}", .{deps_intermediate_path}));
+    run_angle_build.addPrefixedFileArg("--orca-tool=", orca_tool_exe.getEmittedBin());
+    RunHelpers.addPythonArg(run_angle_build, target, b);
+    RunHelpers.addCmakeArg(run_angle_build, target, b);
+
+    const build_angle_step = b.step("angle", "Build Angle libs");
+    build_angle_step.dependOn(&run_angle_build.step);
+
+    /////////////////////////////////////////////////////////
+    // dawn build
+
+    var run_dawn_build: *Build.Step.Run = b.addRunArtifact(build_deps_exe);
+    run_dawn_build.addArg("--lib=dawn");
+    run_dawn_build.addArg(b.fmt("--sha={s}", .{shas.dawn}));
+    run_dawn_build.addArg(b.fmt("--intermediate={s}", .{deps_intermediate_path}));
+    run_dawn_build.addPrefixedFileArg("--orca-tool=", orca_tool_exe.getEmittedBin());
+    RunHelpers.addPythonArg(run_dawn_build, target, b);
+    RunHelpers.addCmakeArg(run_dawn_build, target, b);
+
+    const build_dawn_step = b.step("dawn", "Build Dawn libs");
+    build_dawn_step.dependOn(&run_dawn_build.step);
+
+    /////////////////////////////////////////////////////////
+    // CI helper to print the desired angle/dawn shas to stdout
```

Comment:
Make this comment better - specify it's needed to determine the appropriate sha in build-dawn-<platform>.yml